cmake/tpl/SundialsCaliper.cmake

```diff
@@ -56,28 +53,15 @@ if(CALIPER_FOUND AND (NOT CALIPER_WORKS))
   set(CALIPER_TEST_DIR ${PROJECT_BINARY_DIR}/CALIPER_TEST)
   file(MAKE_DIRECTORY ${CALIPER_TEST_DIR})
 
-  # Create a CMakeLists.txt file
-  file(
-    WRITE ${CALIPER_TEST_DIR}/CMakeLists.txt
-    "cmake_minimum_required(VERSION ${CMAKE_VERSION})\n"
-    "project(ltest C)\n"
-    "set(CMAKE_VERBOSE_MAKEFILE ON)\n"
-    "set(CMAKE_BUILD_TYPE \"${CMAKE_BUILD_TYPE}\")\n"
-    "set(CMAKE_C_COMPILER \"${CMAKE_C_COMPILER}\")\n"
-    "set(CMAKE_C_STANDARD \"${CMAKE_C_STANDARD}\")\n"
-    "set(CMAKE_C_FLAGS \"${CMAKE_C_FLAGS}\")\n"
-    "set(CMAKE_C_FLAGS_RELEASE \"${CMAKE_C_FLAGS_RELEASE}\")\n"
-    "set(CMAKE_C_FLAGS_DEBUG \"${CMAKE_C_FLAGS_DEBUG}\")\n"
-    "set(CMAKE_C_FLAGS_RELWITHDEBUGINFO \"${CMAKE_C_FLAGS_RELWITHDEBUGINFO}\")\n"
-    "set(CMAKE_C_FLAGS_MINSIZE \"${CMAKE_C_FLAGS_MINSIZE}\")\n"
-    "add_executable(ltest ltest.c)\n"
-    "target_include_directories(ltest PRIVATE \"${caliper_INCLUDE_DIR}\")\n"
-    "target_link_libraries(ltest \"-L${caliper_LIB_DIR}\")\n"
-    "target_link_libraries(ltest caliper)\n")
+  if(CXX_FOUND)
+    set(_ext cpp)
+  else()
+    set(_ext c)
+  endif()
```

Comment:
Why does this need to be a C or C++ file?

Reply:
There seems to be a bug (or at least a strange behavior when using `nvhpc`) with `try_compile` where if C++ is enabled in the parent project, the `try_compile` subproject fails with

```
CMake Error in CMakeLists.txt:
  No known features for CXX compiler

  ""

  version .
```

but if you build `ltest` as a C++ code then it identifies the features fine.

Reply:
Interesting, it would be good to add a comment about this issue for future reference